spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteMergeInto.scala

```diff
@@ -215,5 +227,41 @@ case class RewriteMergeInto(spark: SparkSession) extends Rule[LogicalPlan] with
     }
     !(actions.size == 1 && hasUnconditionalDelete(actions.headOption))
   }
+
+  def buildWritePlan(
+     childPlan: LogicalPlan,
+     table: Table): LogicalPlan = {
+    table match {
+      case iceTable: SparkTable =>
+        val numShufflePartitions = spark.sessionState.conf.numShufflePartitions
+        val table = iceTable.table()
+        val distributionMode: String = table.properties
+          .getOrDefault("write.distribution-mode", TableProperties.WRITE_DISTRIBUTION_MODE_RANGE)
+        val order = toCatalyst(toOrderedDistribution(table.spec(), table.sortOrder(), true), childPlan)
+        distributionMode.toLowerCase(Locale.ROOT) match {
```

Comment:
The Flink commit also added a Java enum for this. We could use that instead of string matching here. It handles case insensitive mapping, too: `DistributionMode.fromName(modeName)`